OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc

```diff
@@ -455,8 +457,89 @@ fmi2Component fmi2Instantiate(fmi2String instanceName, fmi2Type fmuType, fmi2Str
 
   /* intialize modelData */
   fmu2_model_interface_setupDataStruc(comp->fmuData, comp->threadData);
-  useStream[LOG_STDOUT] = 1;
-  useStream[LOG_ASSERT] = 1;
+
+  char* flags_filename = malloc(strlen(fmuResourceLocation) + 10);
+  flags_filename[0] = 0;
+  strcat(flags_filename, fmuResourceLocation);
+  strcat(flags_filename, "/flags");
+
+  char* flags = 0;
+  long length;
+
+  FILE* file = fopen(flags_filename, "rb");
+
+  if (file)
+  {
+    fseek(file, 0, SEEK_END);
+    length = ftell(file)+1;
+    fseek(file, 0, SEEK_SET);
+    flags = malloc(length*sizeof(char));
+    if (flags)
+    {
+      fread(flags, 1, length, file);
+      strcat(flags, "\n"); // just to make things easier
+    }
+    fclose(file);
+  }
+  else
+    flags = "\n";
+
+  if (!flags)
+    printf("Some error message");
+
+  // count flags (skip empty lines)
+  int argc = 1;
+  for (int i=0, j=1, start=0; flags[i]; ++i)
+    if (flags[i] == '\n')
+    {
+      if (i-start > 0)
+        argc++;
+      start=i+1;
+    }
+
+  // extract all flags
+  char** argv = malloc(argc*sizeof(char*));
+  argv[0] = "<fmu>";
+  for (int i=0, j=1, start=0; flags[i]; ++i)
+    if (flags[i] == '\n')
+    {
+      if (i-start > 0)
+      {
+        argv[j] = malloc((i-start+1)*sizeof(char));
+        memcpy(argv[j], &flags[start], i-start);
+        argv[j][i-start] = '\0';
+        j++;
+      }
+      start=i+1;
+    }
+
+  initDumpSystem();
+  if (helpFlagSet(argc, argv) || checkCommandLineArguments(argc, argv))
+  {
+    infoStreamPrint(LOG_STDOUT, 1, "usage: %s", argv[0]);
+    for (int i=1; i<FLAG_MAX; ++i)
+    {
+      if(FLAG_TYPE[i] == FLAG_TYPE_FLAG)
+      {
+        infoStreamPrint(LOG_STDOUT, 0, "<-%s>\n  %s", FLAG_NAME[i], FLAG_DESC[i]);
+      }
+      else if(FLAG_TYPE[i] == FLAG_TYPE_OPTION)
+      {
+        infoStreamPrint(LOG_STDOUT, 0, "<-%s=value> or <-%s value>\n  %s", FLAG_NAME[i], FLAG_NAME[i], FLAG_DESC[i]);
+      }
+      else
+      {
+        warningStreamPrint(LOG_STDOUT, 0, "[unknown flag-type] <-%s>", FLAG_NAME[i]);
+      }
+    }
+    messageClose(LOG_STDOUT);
+    EXIT(0);
```

Comment:
Why would you ever call EXIT from an FMU?

Reply:
Fair question. That is a code snipped from the original runtime implementation, i guess we should put a warning here instead since it only indicates wrong/ambiguous flags.

Reply:
Or return an FMI failure status code

Reply:
Well, FMI failure code would be the way to go. But that is not really the issue with this PR right now.